test/unit/util_spec.js

```diff
@@ -136,6 +136,19 @@ describe('util', function() {
     });
   });
 
+  describe('log2', function() {
+    it('handles values smaller than/equal to zero', function() {
+      expect(log2(0)).toEqual(0);
+      expect(log2(-1)).toEqual(0);
+    });
+
+    it('handles values larger than zero', function() {
+      expect(log2(1)).toEqual(0);
+      expect(log2(2)).toEqual(1);
+      expect(log2(3)).toEqual(2);
```

Comment:
Since our `log2` function doesn't behave like you would expect, based on the name, how about adding a test-case where the input is a floating point number?

Reply:
Good suggestion! Done in the new commit.